lib/Sema/CSSimplify.cpp

```diff
@@ -3825,8 +3826,41 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
 
     // FIXME(diagnostics): If there were no viable results, but there are
     // unviable ones, we'd have to introduce fix for each specific problem.
-    if (!result.UnviableCandidates.empty())
+    if (!result.UnviableCandidates.empty()) {
+      // Check if we have unviable candidates whose reason for rejection
+      // is either UR_InstanceMemberOnType or UR_TypeMemberOnInstance
+      auto unviable = [&](const std::pair<OverloadChoice,
```

Comment:
`unviable` is not a very good name for this closure, just write it inline since it's a single use anyway.

Reply:
Yeah, I moved it out as the code looked a bit weird visually. I’ll revert it to being inline like before.